### PR TITLE
Sane gitignore added

### PR DESCRIPTION
**Pull request by illabo**
Fixed gitignore with https://www.gitignore.io/api/xcode,swift and applied some additional settings.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,119 @@
...
+# *.[oa]
+# *~
+
+# Created by https://www.gitignore.io/api/xcode,swift
+# Edit at https://www.gitignore.io/?templates=xcode,swift
+
+### Swift ###
+# Xcode
+#
+# gitignore contributors: remember to update Global/Xcode.gitignore, Objective-C.gitignore & Swift.gitignore
```
Comment (evnik) [inline .gitignore:14]: Please cleanup these notes (headers might be kept)

File: .gitignore [hunk 2]
```diff
@@ -1 +1,119 @@
...
+
+## Playgrounds
+timeline.xctimeline
+playground.xcworkspace
+
+# Swift Package Manager
+# Add this line if you want to avoid checking in source code from Swift Package Manager dependencies.
+# Packages/
+# Package.pins
+# Package.resolved
```
Comment (evnik) [inline .gitignore:50]: Let's uncomment it and remove the note

File: .gitignore [hunk 3]
```diff
@@ -1 +1,119 @@
...
+.build/
+
+# CocoaPods
+# We recommend against adding the Pods directory to your .gitignore. However
+# you should judge for yourself, the pros and cons are mentioned at:
+# https://guides.cocoapods.org/using/using-cocoapods.html#should-i-check-the-pods-directory-into-source-control
```
Comment (evnik) [inline .gitignore:56]: Please remove the note

File: .gitignore [hunk 4]
```diff
@@ -1 +1,119 @@
...
+Pods/
+# Add this line if you want to avoid checking in source code from the Xcode workspace
+# *.xcworkspace
```
Comment (evnik) [inline .gitignore:59]: I'm not sure in what case we might need it, so I'd suggest to remove these lines as well

File: .gitignore [hunk 5]
```diff
@@ -1 +1,119 @@
...
+
+# Pod project
+Gemfile.lock
+.ruby-version
+Example/Podfile.lock
+Example/Pods/Local Podspecs/*podspec.json
+Example/Pods/Manifest.lock
```
Comment (evnik) [inline .gitignore:66]: As I understand, these lines might be removed, because we have `Pods/` ignored

File: .gitignore [hunk 6]
```diff
@@ -1 +1,119 @@
...
+
+# Carthage
+# Add this line if you want to avoid checking in source code from Carthage dependencies.
+# Carthage/Checkouts
```
Comment (evnik) [inline .gitignore:70]: Let's uncomment it and remove the note

File: .gitignore [hunk 7]
```diff
@@ -1 +1,119 @@
...
+
+Carthage/Build
+
+# fastlane
+# It is recommended to not store the screenshots in the git repo. Instead, use fastlane to re-generate the
+# screenshots whenever they are needed.
+# For more information about the recommended setup visit:
+# https://docs.fastlane.tools/best-practices/source-control/#source-control
```
Comment (evnik) [inline .gitignore:78]: Please remove the note

File: .gitignore [hunk 8]
```diff
@@ -1 +1,119 @@
...
+fastlane/Preview.html
+fastlane/screenshots/**/*.png
+fastlane/test_output
+screenshots/
+
+# Code Injection
+# After new code Injection tools there's a generated folder /iOSInjectionProject
+# https://github.com/johnno1962/injectionforxcode
+
+iOSInjectionProject/
```
Comment (evnik) [inline .gitignore:90]: Is this still useful?

File: .gitignore [hunk 9]
```diff
@@ -1 +1,119 @@
...
+
+### Xcode Patch ###
+*.xcodeproj/*
+!*.xcodeproj/project.pbxproj
+!*.xcodeproj/xcshareddata/
+!*.xcworkspace/contents.xcworkspacedata
+/*.gcno
+**/xcshareddata/WorkspaceSettings.xcsettings
+
+# End of https://www.gitignore.io/api/xcode,swift
```
Comment (evnik) [inline .gitignore:110]: Please cleanup notes

File: .gitignore [hunk 10]
```diff
@@ -1 +1,119 @@
...
+
+# xcodegen
+# *.xcodeproj
```
Comment (evnik) [inline .gitignore:113]: I'm not sure in what case we might need it, so I'd suggest to remove these lines as well

**Review by evnik — COMMENTED** [review]
Looks good, but let's cleanup notes and irrelevant parts, so it'll be easier to maintain

File: .gitignore (hunk 8 above)
Comment (illabo) [reply .gitignore:90]: No way

File: .gitignore (hunk 5 above)
Comment (illabo) [reply .gitignore:66]: It's for the projects which are Pods themself.
However I'd keep `Gemfile.lock` and `.ruby-version` as these files may appear if some tools in ruby are used (even if it is unlikely we use it on this project).

**Review by evnik — APPROVED** [review]

**Pull request merged**